article/_posts/2019-11-21-stealml.md

```diff
@@ -0,0 +1,87 @@
+
+For all the datasets shown in table 3, the authors could train a model and extract it in d+1 predictions with 100% confidence error. For these models, the attack only required 41 queries on average and 113 at most.
+
+The authors also took a shot a attacking multiclass LR (MLR) models, such as softmax and one-vs-rest (OvR) models. MLRs models are parametrized by $$w \in R^{cd}$$, and no longer provide linear equation systems to be solved. However, by minimizing an appropriate loss function, such as the logistic loss, the optimization problem becomes convex and can converge to a global minimum. This also applies to multi-layer perceptrons (MLP) which are parametrized by $$w \in R^{dh+hc}$$ where $$h$$ is the number of hidden layers. However in the case of MLPs, the loss function is no longer convex and the optimization may not converge to a global minimum.
+
+
+![](/article/images/stealml/tab4.jpeg)
+
+## Online Model Extraction Attacks
+
+The authors then tried to extract models setup online on BigML and Amazon AWS. BigML uses decision tree which will not be adressed in this review. Amazon, however, allows users to upload their data and then Amazon will perform feature extraction and then train a model on these features.
+
+The problem now is to find what features were extracted and which model was used. However, AWS does support "partial input", where you can send an image for prediction which does not have some of the extracted features and still get a result. Without going into details, the authors were able to reverse-engineer the features and still get a copy of the model with 100% matches.
+
+![](/article/images/stealml/tab7.jpeg)
+
+In all the attacks, the authors were able to learn about the model's class, the enable feature extractors and hyperparameters through the ML service AWS's documentation. In a real world scenario, an attacker could narrow down its hyperparameter search through similar techniques and cross-validation.
+
+## Extraction Given Class Labels Only
+
+Without going into details, the authors wanted to see if they could extract model through only class predictions. The general strategy is called retraining, where the authors query the models with data points that are close to $$\hat{f}$$'s decision boundary.
+
+![](/article/images/stealml/fig4.jpeg)
+![](/article/images/stealml/fig5.jpeg)
+![](/article/images/stealml/fig6.jpeg)
+
+where $$\alpha$$ is the budget, or multiplier of $$(d+1)$$ data points used.
+
+##  Prevention from such attacks
+
+The authors suggest some countermeasures to such attacks, such as rounding off the prediction's accuracy. The results of round the accuracy (or only providing class labels is reported below)
```

Comment:
The results of **rounding** the accuracy (or only providing class labels) is reported below.

Reply:
Woops, fixed in 4ef3ccc